### PR TITLE
Allow two call input expressions to use the same outer lookup. Closes #2973

### LINKED ISSUE #2973: Conditional task with same input twice fails
```wdl
task two_input_task {
    String inA
    String inB
    command { echo ${inA} and ${inB} }
    runtime { docker: "ubuntu" }
}

workflow duplicate_conditional_input {
    String my_input = "input"
    if (false) {
        call two_input_task { input: inA = my_input, inB = my_input }
    }
}
```

Runs in 29 (d45f001). In develop (0fd1635) this crashes with:

```java
[2017-12-02 16:30:57,99] [info] Retrieved 1 workflows from the WorkflowStoreActor
[2017-12-02 16:30:59,03] [error] WorkflowManagerActor Workflow e7ef9b84-2a08-4295-a2ab-fea19ddcba76 failed (during MaterializingWorkflowDescriptorState): Workflow input processing failed:
Unable to build WOM node for If '$if_0': Two or more nodes have the same FullyQualifiedName: ^.my_input
cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor$$anon$1: Workflow input processing failed:
Unable to build WOM node for If '$if_0': Two or more nodes have the same FullyQualifiedName: ^.my_input
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor.cromwell$engine$workflow$lifecycle$materialization$MaterializeWorkflowDescriptorActor$$workflowInitializationFailed(MaterializeWorkflowDescriptorActor.scala:211)
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor$$anonfun$2.applyOrElse(MaterializeWorkflowDescriptorActor.scala:181)
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor$$anonfun$2.applyOrElse(MaterializeWorkflowDescriptorActor.scala:176)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:34)
	at akka.actor.FSM.processEvent(FSM.scala:665)
	at akka.actor.FSM.processEvent$(FSM.scala:662)
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor.akka$actor$LoggingFSM$$super$processEvent(MaterializeWorkflowDescriptorActor.scala:136)
	at akka.actor.LoggingFSM.processEvent(FSM.scala:801)
	at akka.actor.LoggingFSM.processEvent$(FSM.scala:783)
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor.processEvent(MaterializeWorkflowDescriptorActor.scala:136)
	at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:659)
	at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:653)
	at akka.actor.Actor.aroundReceive(Actor.scala:514)
	at akka.actor.Actor.aroundReceive$(Actor.scala:512)
	at cromwell.engine.workflow.lifecycle.materialization.MaterializeWorkflowDescriptorActor.aroundReceive(MaterializeWorkflowDescriptorActor.scala:136)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
	at akka.actor.ActorCell.invoke(ActorCell.scala:496)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### PR DESCRIPTION
**Pull request by cjllanwarne**
The `WdlCall` expression handler contained a `traverse` which generated OGINs for outer lookups in parallel.
To avoid that, the PR pre-generates the OGINs for the expressions to use and doesn't let the `traverse` create its own.

### CODE REVIEW

**Comment by cjllanwarne** [comment]
New WomGraph for the `scatter` example in #2973:

test

File: wdl/src/main/scala/wdl/WdlCall.scala [hunk 1]
```diff
@@ -82,17 +83,21 @@ object WdlCall {
...
-          case _: WdlTaskCall => TaskCallInputExpressionNode.apply _
-          case _ => PlainAnonymousExpressionNode.apply _
-        }
-
-        WdlWomExpression.toAnonymousExpressionNode(identifier, WdlWomExpression(wdlExpression, wdlCall), localLookup, outerLookup, preserveIndexForOuterLookups, wdlCall, constructor) map {
-          LocalName(inputName) -> _
-        }
+    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = {
+      val precomputedOgins: Map[String, OutputPort] = outerLookup.filterNot { ol => localLookup.contains(ol._1) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }
+      val newLocalLookup = localLookup ++ precomputedOgins
```
Comment (cjllanwarne) [inline wdl/src/main/scala/wdl/WdlCall.scala]: These are the only new lines (and using `newLocalLookup` below). The rest is reformatting to fit inside the `def`'s new `{` curlies `}`

File: wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala [hunk 2]
```diff
@@ -185,4 +186,49 @@ object WdlNestedConditionalWomSpec {
       |    Int? m1_out = m1.out
       |  }
       |}""".stripMargin ++ taskMirror
+
+  val doubleNestedVariableReferences =
+    """|workflow double_nested_var_refs {
+       |
+       |  Int i = 27
+       |  Int j = 32
+       |  Int k = 56
```
Comment (kshakir) [inline wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala]: Curious, are these your lucky lottery numbers?

File: wdl/src/main/scala/wdl/WdlCall.scala [hunk 3]
```diff
@@ -82,17 +83,21 @@ object WdlCall {
...
-        val constructor = wdlCall match {
-          case _: WdlTaskCall => TaskCallInputExpressionNode.apply _
-          case _ => PlainAnonymousExpressionNode.apply _
-        }
-
-        WdlWomExpression.toAnonymousExpressionNode(identifier, WdlWomExpression(wdlExpression, wdlCall), localLookup, outerLookup, preserveIndexForOuterLookups, wdlCall, constructor) map {
-          LocalName(inputName) -> _
-        }
+    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = {
+      val precomputedOgins: Map[String, OutputPort] = outerLookup.filterNot { ol => localLookup.contains(ol._1) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }
```
Comment (kshakir) [inline wdl/src/main/scala/wdl/WdlCall.scala]: Someday we'll be as evolved as [these guys](https://github.com/broadinstitute/clio/pull/182).

For now:

`val precomputedOgins: Map[String, OutputPort] = outerLookup filterNot { case (name, _) => localLookup.contains(name) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }`

Or:

```scala
val precomputedOgins: Map[String, OutputPort] = outerLookup filterNot {
  case (name, _) => localLookup.contains(name)
} map {
  case (name, port) =>
    name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort
}
```

File: wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala [hunk 4]
```diff
@@ -185,4 +186,49 @@ object WdlNestedConditionalWomSpec {
...
+       |    echo $((${i} * ${j}))
+       |  }
+       |  output {
+       |    Int out = read_int(stdout())
+       |  }
+       |  runtime {
+       |    docker: "ubuntu:latest"
+       |  }
+       |}""".stripMargin
 }
```
Comment (kshakir) [inline wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala]: While you're here, slip a newline in for me as a holiday gift. 🕎🎄

File: wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala (hunk 2 above)
Comment (mcovarr) [reply wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala]: He's spending his weekend writing tests for WOM bugs so I'm guessing not.

Comment (cjllanwarne) [reply wdl/src/test/scala/wom/WdlNestedConditionalWomSpec.scala]: 😢

**Pull request merged**